### DISCUSSION TITLE
How to skip executions using PauseTrigger ResumeTrigger

### DISCUSSION THREAD

**Discussion by ghost** [quartznet/quartznet#1637] [Q&A]
Hi,

My goal is to be able to "disable" triggers temporarily and then re-enable them again at a later time. I want the scheduler to skip (not run) any of the executions that it would have run, during the time it is disabled.

I have created this trigger just for testing:

```csharp
ITrigger dailyTrigger = TriggerBuilder.Create()
    .WithIdentity("Every15Seconds", "DailyTriggerGroup")
    .ForJob(jobKey)
    .UsingJobData(jobDataMap)
    .WithDailyTimeIntervalSchedule(x => x
        .WithIntervalInSeconds(15)
        .StartingDailyAt(new TimeOfDay(DateTime.Now.Hour, DateTime.Now.AddMinutes(1).Minute))
        .EndingDailyAfterCount(15)
        .OnMondayThroughFriday()
        .WithMisfireHandlingInstructionDoNothing()
        .InTimeZone(TimeZoneInfo.FindSystemTimeZoneById("Eastern Standard Time"))
        )
    .StartNow()
    .WithDescription("Every 15 Seconds")
    .Build();
```

After I create the trigger and it is running, I call the following code to pause it:

```csharp
Task<IScheduler> sched = new StdSchedulerFactory(this.properties).GetScheduler();
IScheduler s = await sched;
await s.PauseTrigger(new TriggerKey(trig.TriggerName, trig.TriggerGroup));
```

After a minute or so, I re-enable it using the following code:

```csharp
Task<IScheduler> sched = new StdSchedulerFactory(this.properties).GetScheduler();
IScheduler s = await sched;
await s.ResumeTrigger(new TriggerKey(trig.TriggerName, trig.TriggerGroup));
```

**MY ISSUE:**
When the trigger resumes, the scheduler fires all of the executions that would have fired during the time that it was paused. Since I'm wanting an "enable / disable" feature in our app, I wanted it to skip the executions when it is paused (disabled).

I was hoping that ".WithMisfireHandlingInstructionDoNothing()" would prevent the "missed" fires, from firing but it doesn't.

Is there a better way of doing what I'm trying to do? I guess I'll experiment with removing EndDailyAfterCount(n), in case that is affecting it.

**Comment by CrazyMoka** [comment]
You find a solution for this?
I'm going to be in the same boat. Trying to decide to pause the JOB or pause the Trigger as many can belong to the same job.
